internal/pkg/application/transform/transformers.go

```diff
@@ -241,71 +353,117 @@ func WaterConsumptionObserved(ctx context.Context, msg iotcore.MessageAccepted,
+			p := append(props, w)
+			err := storeWaterConsumption(ctx, log, cbClient, entityID, p...)
 			if err != nil {
 				return err
 			}
 		}
 	}
 
-	// then continue with delta volumes
-	for _, record := range msg.Pack {
-		if strings.EqualFold("DeltaVolume", record.Name) {
-			vol := math.Floor((*record.Sum + 0.0005) * 1000)
-			dt := time.Unix(int64(record.Time), 0).UTC().Format(time.RFC3339Nano)
-			props := waterConsumptionProps(vol, dt, observedBy, msg)
+	return nil
+}
 
-			err := storeWaterConsumption(ctx, log, cbClient, entityID, props...)
-			if err != nil {
-				return err
-			}
+func GreenspaceRecord(ctx context.Context, msg core.MessageAccepted, cbClient client.ContextBrokerClient) error {
+	curDateTime := msg.Timestamp
+	if cdt, ok := msg.GetString("CurrentDateTime"); ok {
+		if idx := strings.Index(cdt, "."); idx > 0 {
+			curDateTime = cdt[0:idx] + "Z"
 		}
 	}
 
-	// last add current volume if exists
-	if currentVolume, ok := msg.GetFloat64("CurrentVolume"); ok {
-		if currentDateTime, ok := msg.GetTime("CurrentDateTime"); ok {
-			vol := math.Floor((currentVolume + 0.0005) * 1000)
-			dt := time.Unix(int64(currentDateTime), 0).UTC().Format(time.RFC3339Nano)
-			props := waterConsumptionProps(vol, dt, observedBy, msg)
+	entityID := fmt.Sprintf("%s%s", "urn:ngsi-ld:GreenspaceRecord:", msg.Sensor)
+	observedBy := fmt.Sprintf("%s%s", fiware.DeviceIDPrefix, msg.Sensor)
+
+	headers := map[string][]string{"Content-Type": {"application/ld+json"}}
 
-			err := storeWaterConsumption(ctx, log, cbClient, entityID, props...)
+	buildfragment := func(patchProperties ...entities.EntityDecoratorFunc) error {
+		fragment, err := entities.NewFragment(patchProperties...)
+		if err != nil {
+			return fmt.Errorf("entities.NewFragment failed: %w", err)
+		}
+
+		properties := append(patchProperties, DateObserved(curDateTime))
+
+		//_, err = cbClient.UpdateEntityAttributes(ctx, entityID, fragment, headers)
+		_, err = cbClient.MergeEntity(ctx, entityID, fragment, headers)
+
+		if err != nil {
+			// If we failed to update the entity's attributes, we need to create it
+			properties := append(properties, entities.DefaultContext())
+
+			if msg.HasLocation() {
+				properties = append(properties, Location(msg.Latitude(), msg.Longitude()))
+			}
+
+			var entity types.Entity
+			entity, err = entities.New(entityID, fiware.GreenspaceRecordTypeName, properties...)
 			if err != nil {
-				return err
+				return fmt.Errorf("entities.New failed: %w", err)
+			}
+
+			_, err = cbClient.CreateEntity(ctx, entity, headers)
+			if err != nil {
+				err = fmt.Errorf("create entity failed: %w", err)
 			}
 		}
-	} else {
-		log.Debug().Msgf("no volume property was found in message from %s", msg.Sensor)
+
+		return err
 	}
 
-	return nil
-}
+	// GreenspaceRecord is called by one of its properties. First out creates the entity, all other subsequent calls, independent which property, updates the entity.
```

Comment:
I'm a little unsure of what this comment means, but I think it should probably say "GreenspaceRecord is called by one of its properties. First it creates the entity, and on all subsequent calls, independent of which property, it updates the entity."